### PR TITLE
Java 9 Modules Smoketest

### PR DESCRIPTION
**Pull request by randomanderson**
This adds a smoketest with the following characteristics:
- A modularized app with no requirements.
- For each installed jdk, a minimal jdk image is created with `java.base`, `java.instrument`, and the modularized app as the only modules

The modularized app is simple because at present, the Datadog api is not friendly to being imported.  This will be addressed in a separate pull request.

Lastly, 2 minor bugs are fixed with `DriverInstrumentation` and `ConnectionInstrumentation`.  Without the `java.sql` module available, the application crashed on startup.

### CODE REVIEW

File: dd-smoke-tests/java9-modules/java9-modules.gradle [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+ext {
+  minJavaVersionForTests = JavaVersion.VERSION_1_9
+}
+
+apply from: "${rootDir}/gradle/java.gradle"
+
+sourceCompatibility = "1.9"
+targetCompatibility = "1.9"
```
Comment (tylerbenson) [inline dd-smoke-tests/java9-modules/java9-modules.gradle]: With my Play PR, this won't be necessary.

File: dd-smoke-tests/java9-modules/java9-modules.gradle [hunk 2]
```diff
@@ -0,0 +1,35 @@
+plugins {
+  id 'org.beryx.jlink' version '2.1.1' // latest version that doesn't have a Java 11+ warning
+}
+
+ext {
+  minJavaVersionForTests = JavaVersion.VERSION_1_9
```
Comment (tylerbenson) [inline dd-smoke-tests/java9-modules/java9-modules.gradle]: Interesting... I didn't expect we could skip Java 8 with this config.

File: dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/ConnectionInstrumentation.java [hunk 3]
```diff
@@ -43,7 +43,7 @@ public ConnectionInstrumentation() {
         nameStartsWith("prepare")
             .and(takesArgument(0, String.class))
             // Also include CallableStatement, which is a sub type of PreparedStatement
-            .and(returns(safeHasSuperType(named(PreparedStatement.class.getName())))),
+            .and(returns(safeHasSuperType(named("java.sql.PreparedStatement")))),
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/ConnectionInstrumentation.java]: Nice find.

File: settings.gradle [hunk 4]
```diff
@@ -23,6 +23,7 @@ include ':dd-smoke-tests:play'
 include ':dd-smoke-tests:springboot'
 include ':dd-smoke-tests:wildfly'
 include ':dd-smoke-tests:cli'
+include ':dd-smoke-tests:java9-modules'
```
Comment (tylerbenson) [inline settings.gradle]: Small nit... please sort this section alphabetically.

**Comment by randomanderson** [comment]
I may have a solution to the compiling issue.  I'll wait until after the Play pull request is merged so there aren't multiple gradle changes pending

**Pull request merged**